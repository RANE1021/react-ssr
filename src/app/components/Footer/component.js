import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";

const Footer = ({ styles }) => (
  <div className={styles.wrapper}>
    <NavLink to="/">Home</NavLink>
    <NavLink to="/survey">Survey</NavLink>
    <NavLink to="/about">About</NavLink>
  </div>
);

Footer.propTypes = {
  styles: PropTypes.object.isRequired
};

export default Footer;
