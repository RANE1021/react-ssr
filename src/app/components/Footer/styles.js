export const wrapper = () => ({
  display: "flex",
  justifyContent: "space-evenly",
  height: "18",
  fontColor: "#1C1CD8",
  margin: "auto",
  paddingTop: "10px",
  paddingLeft: "5px",
  paddingRight: "5px"
});
