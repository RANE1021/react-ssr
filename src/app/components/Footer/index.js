import { connect } from "react-fela";

import * as styles from "./styles";
import component from "./component";

export default connect(styles)(component);
