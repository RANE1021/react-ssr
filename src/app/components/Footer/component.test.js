import React from "react";
import { shallow } from "enzyme";

import Footer from "./component";

const styles = {};

describe("The footer component mounts with styles and a div", () => {
  const shallowFooter = shallow(<Footer styles={styles} />);
  it("loads with a div", () => {
    expect(shallowFooter.find("div").length).toBe(1);
  });

  it("loads with 3 NavLinks", () => {
    expect(shallowFooter.find("NavLink").length).toBe(3);
  });
});
