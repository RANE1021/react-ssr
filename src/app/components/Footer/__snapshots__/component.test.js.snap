// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The footer component mounts with styles and a div matches previous snapshot 1`] = `[Function]`;
